src/test/javascript/general.controller.test.js

```diff
@@ -228,63 +438,133 @@ describe("GeneralController", function () {
         });
     });
 
+    describe("launchImportModal", () => {
+        it("should set listName to listName passed in", () => {
+            scope.listName = "";
+            scope.launchImportModal("admin");
+            expect(scope.listName).toBe("admin");
+        });
+
+        it("should check that the import modal is launched", () => {
+            spyOn(uibModal, "open").and.callThrough();
+            scope.launchImportModal(scope.listName);
+            expect(uibModal.open).toHaveBeenCalled();
+        });
+    });
+
+    describe("addMembers", () => {
+        it("should set listName to the listName that we are passing in", () => {
+            scope.listName = "admin";
+            scope.addMembers(scope.listName);
+            expect((scope.listName)).toBe("admin");
+        });
+
+        it("should make scope.emptyInput true if scope.manageMembers is empty", () => {
+            scope.emptyInput = false;
+            scope.manageMembers = null;
+            scope.addMembers(scope.listName);
+            expect(scope.emptyInput).toBeTrue();
+        });
+
+        it("should call addMultipleMembers when the usersToAdd is below our maxImport", () => {
+            spyOn(scope, "addMultipleMembers");
+            scope.manageMembers = "iamtst01 iamtst02";
+            scope.addMembers(scope.listName);
+            expect(scope.addMultipleMembers).toHaveBeenCalled();
+        });
+
+        it("should call launchDynamicModal when the usersToAdd is above the multiAddThreshold", () => {
+            spyOn(scope, "launchDynamicModal");
+            let arr = [];
+            for (let i = 0; i < 102; i++) {
+                arr.push("iamtst01");
+            }
+            scope.manageMembers = arr.toString().split(",").join(" ");
+            scope.addMembers(scope.listName);
+            expect(scope.launchDynamicModal).toHaveBeenCalled();
+        });
+
+        it("should call launchDynamicModal when the members we are adding are above the maxImport", () => {
+            spyOn(scope, "launchDynamicModal");
+            let arr = [];
+            for (let i = 0; i < 100002; i++) {
+                arr.push("iamtst01");
+            }
+            scope.manageMembers = arr.toString().split(",").join(" ");
+            scope.addMembers(scope.listName);
+            expect(scope.launchDynamicModal).toHaveBeenCalled();
+        });
+
+        it("should call validateAndAddUser when numMembers is less than 0", () => {
+            spyOn(scope, "validateAndAddUser");
+            scope.manageMembers = "iamtst01";
+            scope.addMembers(scope.listName);
+            expect(scope.manageMembers).toBe("iamtst01");
+            expect(scope.validateAndAddUser).toHaveBeenCalled();
+        });
+    });
+
     describe("sanitizer", () => {
         let goodFile, badFile, parseFile;
         beforeEach(() => {
-            let bad1, bad2, bad3, bad4, bad5, bad6, bad7, bad8, bad9, bad10,bad11, bad12, bad13, bad14, bad15, bad16, bad17;
```

Comment:
Can these bad variables be implemented better? This looks a bit clunky. Maybe a separate file? I am not sure what could be done.